MODULE.bazel

```diff
@@ -83,6 +80,33 @@ use_repo(
     "rules_haskell_asterius_build_setting",
 )
 
+haskell_toolchains = use_extension(
+    "@rules_haskell//extensions:haskell_toolchains.bzl",
+    "haskell_toolchains",
+)
+
+haskell_toolchains.bindists()
+
+use_repo(
+    haskell_toolchains,
+    "all_bindist_toolchains",
+    "rules_haskell_python_local",
+    "rules_haskell_sh_posix_local",
+)
+
+# We need to register the nix toolchain before the bindist ones so it
+# take precedence when we use the nixpkgs config.
+register_toolchains(
+    "@rules_haskell_ghc_nixpkgs_toolchain//:toolchain",
+    dev_dependency = True,
+)
+
+register_toolchains(
+    "@all_bindist_toolchains//:all",
+    "@rules_haskell_python_local//:toolchain",
+    "@rules_haskell_sh_posix_local//:all",
+)
```

Comment:
Happy to stick to this for this PR.
In the long run, similarly to https://github.com/tweag/rules_haskell/issues/1915, I think the default POSIX toolchain should be something that rules_sh already registers in its own MODULE.bazel file.
In other words, under bzlmod rules_haskell should only register Haskell toolchains itself (ignoring dev dependencies). Other toolchains should be registered by other modules. (rules_haskell_test can of course still register whatever is needed for testing).

Reply:
I removed the registration of `rules_haskell_sh_posix_local` toolchain and the one from `rules_sh` is used instead :+1: 

The last version of `rules_python` also registers a default toolchain but is does not work when bazel is run as root (see [this issue](https://github.com/bazelbuild/rules_python/issues/1169)), so I kept the `rules_haskell_python_local` for now.